### PR TITLE
[cli] [tabletserver/gc] Migrate gc flags to pflags

### LINKED ISSUE #10983: Switch flag definitions to be on pflag instead of flag in `package go/vt/vttablet/tabletserver/gc`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/vttablet/tabletserver/gc/*.go
go/vt/vttablet/tabletserver/gc/tablegc.go:var checkInterval = flag.Duration("gc_check_interval", 1*time.Hour, "Interval between garbage collection checks")
go/vt/vttablet/tabletserver/gc/tablegc.go:var purgeReentranceInterval = flag.Duration("gc_purge_check_interval", 1*time.Minute, "Interval between purge discovery checks")
go/vt/vttablet/tabletserver/gc/tablegc.go:var gcLifecycle = flag.String("table_gc_lifecycle", "hold,purge,evac,drop", "States for a DROP TABLE garbage collection cycle. Default is 'hold,purge,evac,drop', use any subset ('drop' implcitly always included)")
```

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
## Description

Updates tablegc related flag references specified in #10983

## Related Issue(s)

Fixes #10983

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/tabletserver/gc/tablegc.go [hunk 1]
```diff
@@ -27,6 +26,10 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/spf13/pflag"
+
+	"vitess.io/vitess/go/vt/servenv"
+
```
Comment (ajm188) [inline go/vt/vttablet/tabletserver/gc/tablegc.go:32]: nit: this blank line is unnecessary, can you remove it?

File: go/vt/vttablet/tabletserver/gc/tablegc.go [hunk 2]
```diff
@@ -166,7 +180,7 @@ func (collector *TableGC) Open() (err error) {
 		// already open
 		return nil
 	}
-	collector.lifecycleStates, err = schema.ParseGCLifecycle(*gcLifecycle)
+	collector.lifecycleStates, err = schema.ParseGCLifecycle(gcLifecycle)
 	if err != nil {
 		return fmt.Errorf("Error parsing -table_gc_lifecycle flag: %+v", err)
```
Comment (ajm188) [inline go/vt/vttablet/tabletserver/gc/tablegc.go:185]:
```suggestion
		return fmt.Errorf("Error parsing --table_gc_lifecycle flag: %+v", err)
```

Comment (ajm188) [reply go/vt/vttablet/tabletserver/gc/tablegc.go:185]: (please check for other instances of this, which i meant to clean up previously .. sorry about that!)

File: go/vt/vttablet/tabletserver/gc/tablegc.go (hunk 1 above)
Comment (rohit-nayak-ps) [reply go/vt/vttablet/tabletserver/gc/tablegc.go]: fixed

File: go/vt/vttablet/tabletserver/gc/tablegc.go (hunk 2 above)
Comment (rohit-nayak-ps) [reply go/vt/vttablet/tabletserver/gc/tablegc.go]: done

**Review by ajm188 — APPROVED** [review]

**Pull request merged**